### ISSUE TITLE
Issues with special characters in filenames

### ISSUE THREAD

**Issue by 145818** [NeogitOrg/neogit#871]
### Description

Seems non English characters, such as åäö for instance, can not be used in filenames for neogit.

### Neovim version

NVIM v0.9.2
Build type: Release
LuaJIT 2.1.1694285958

### Operating system and version

Artix Linux

### Steps to reproduce

1. Create a file such as `touch åäö.txt` and open nvim then neogit. The file will show up as untracked and with the name `"\303\245\303\244\303\266.txt"`

2. It's not possible to stage it, you'll get an error.

### Expected behavior

Non English characters in filenames would be OK for neogit.

### Actual behavior

Error when trying to stage the file:

``` sh
git --no-pager -c color.ui=always --no-optional-locks add -- "\303\245\303\244\303\266.txt":

    fatal: pathspec '"\303\245\303\244\303\266.txt"' did not match any files


Open the console for details

Press ENTER or type command to continue      
```

### Minimal config
init.lua
```Lua
require "user.plugins"
require "user.neogit"
```

lua/user/plugins.lua
```Lua
  use {
    "NeogitOrg/neogit",
    requires = {
      "nvim-lua/plenary.nvim",
      "nvim-telescope/telescope.nvim",
      "sindrets/diffview.nvim",
    }
  }
```

lua/user/neogit.lua
```Lua
local neogit = require("neogit")

neogit.setup {}
```

**Comment by wrightjjw** [comment]
This issue may stem from git translating `åäö.txt` to `\303\245\303\244\303\266.txt` in its own messages. Git will report the file as such in a simple `git status` from the command line, no neogit involved at all.

**Comment by 145818** [comment]
git names åäö like that indeed, but with plain git I can commit etc. without errors:
``` sh
$ cd /tmp 
$ mkdir test
$ cd test 
$ git init
Initialized empty Git repository in /tmp/test/.git/
$ touch åäö.txt
$ git diff
$ git status
On branch master

No commits yet

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	"\303\245\303\244\303\266.txt"

nothing added to commit but untracked files present (use "git add" to track)
$ git add åäö.txt 
$ git status     
On branch master

No commits yet

Changes to be committed:
  (use "git rm --cached <file>..." to unstage)
	new file:   "\303\245\303\244\303\266.txt"

$ git commit -m 'this works fine'
[master (root-commit) 391eb03] this works fine
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 "\303\245\303\244\303\266.txt"
$ git status                     
On branch master
nothing to commit, working tree clean
``` 
While if I do the same thing with neogit it fails like I mentioned at the beginning of this issue. magit in Emacs does not have this issue either, which makes me wonder if it's a neogit issue (or misconfiguration on my end for that matter).

**Comment by CKolkey** [comment]
Yea, if Magit and git CLI can manage this, I'm confident its on our end.

**Comment by misaflo** [comment]
I reported a [similar bug in forgit](https://github.com/wfxr/forgit/issues/325), a [workaround](https://github.com/wfxr/forgit/issues/325#issuecomment-1751335268) is to tell git to not quote unusual characters in the pathname with `git config core.quotePath false`.

It fixes the problem for Neogit too.

**Comment by CKolkey** [comment]
Fixed with #1071
